frontend/src/pages/PipelineList.tsx

```diff
@@ -264,4 +271,9 @@ class PipelineList extends Page<{}, PipelineListState> {
   }
 }
 
-export default PipelineList;
+const EnhancedPipelineList: React.FC<PageProps> = props => {
+  const namespace = React.useContext(NamespaceContext);
+  return <PipelineList key={namespace} {...props} namespace={namespace} />;
```

Comment:
Per latest discussion, we can let UI always work with empty namespace and let the UI PR implement this instead. Did I miss anything?

Reply:
I thought to leave the UI functionality which implements the namespace selector, to make it easier for the next UI changes. It's also hard to deliver back-end functionality without breaking the UI

Reply:
@StefanoFioravanzo @yanniszark what do you think?

How soon do you think you can implement the UI change? If it's possible to get into the next Kubeflow release targeting March.15, then it sounds OK to me.

> It's also hard to deliver back-end functionality without breaking the UI

I don't fully understand, my understanding of last discussion was that when UI isn't changed, it will not send resource reference key to backend, so backend will always act on the shared pipelines/versions. The behavior will be backward compatible to current situation, no matter multi-user mode or single-user mode.

So I'd prefer moving the UI changes to a separate mini PR, if you want to save some work for @StefanoFioravanzo.

Reply:
@Bobgy I don't think I will have the bandwidth to implement this before March 15 as I already have some things in my backlog. In the case we do need some changes for the UI not to break, then by all means I will look into it. But I'd like for us not to rush this too much so that we can have a proper discussion around the most suitable UX.

Reply:
@Bobgy @StefanoFioravanzo 
Alright, let's merge this PR without any UI changes. I will remove the changes on the UI, and will handle the three scenarios where the client:
1. Does not send resource references.
2.  It sends a resource reference to the public namespace _""_
3.  It sends a resource reference to the user's namespace _x_